File: Taller1/java/Ordenamiento_1/src/ordenamiento/Seleccion.java
package ordenamiento;

/**
 *
 * @author manuel
 */
public class Seleccion {

    public void seleccion(int A[]) {//metodo

        int i, j, menor, pos, tmp;
        for (i = 0; i < A.length - 1; i++) { // tomamos como menor el primero
            menor = A[i]; // de los elementos que quedan por ordenar
            pos = i; // y guardamos su posición
            for (j = i + 1; j < A.length; j++) { // buscamos en el resto
                if (A[j] < menor) { // del array algún elemento
                    menor = A[j]; // menor que el actual
                    pos = j;
                }
            }
            if (pos != i) { // si hay alguno menor se intercambia
                tmp = A[i];
                A[i] = A[pos];
                A[pos] = tmp;
            }
        }
    }
        public void mostrar(int A[], int tam) {//metodo
        for (int i = 0; i < A.length; i++) {
            System.out.printf(A[i]+"\t");// se imprime el Array
        }
            System.out.println("");//salto
    }

}

File: Taller1/c/ordenamiento/seleccion_c/seleccion.c
#include "stdio.h"

int main(void) {
    int i, j, menor, tmp;
    int A[] = {4, 3, 5, 2, 6, 8, 0, 7, 9, 1};
    int tam = 10;
    for (i = 0; i < tam - 1; i++) {
        menor = A[i];
        int pos = i;
        for (j = i + 1; j < tam; j++) {
            if (A[j] < menor) {
                menor = A[j];
                pos = j;
            }
        }
        if (pos != i) {
            tmp = A[i];
            A[i] = A[pos];
            A[pos] = tmp;
        }
    }
    for (i = 0; i < tam; i++) {
        printf("%d , ", A[i]);
    }
    return 0;
}
